src/js/media/html5.js

```diff
@@ -61,6 +61,9 @@ vjs.Html5 = vjs.MediaTechController.extend({
 
     if (this['featuresNativeTextTracks']) {
       this.on('loadstart', vjs.bind(this, this.hideCaptions));
+    } else {
+      // allow us to view the in-band metadata tracks from our shimmed TextTrackList
+      this.on('loadedmetadata', vjs.bind(this, this.addInBandMetadataTracks));
```

Comment:
An in-band metadata track might not be evident until a decent amount of buffering has occurred. If there were a way to make the track registration a bit more robust for that case, that would be great.